### PR TITLE
Add systemd collector

### PR DESCRIPTION
**Pull request by gmlexx**
Send 1 if configured systemd service status is running and 0 otherwise

### CODE REVIEW

File: src/collectors/systemd/systemd-collector.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        Returns the default collector settings
+        """
+        config = super(SystemdCollector, self).get_default_config()
+        config.update({
+            'services': '',
+        })
+        return config
+
+    def collect(self):
+
```
Comment (jaingaurav) [inline src/collectors/systemd/systemd-collector.py]: More of a style thing, please remove this empty line

File: src/collectors/systemd/systemd-collector.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        manager = None
+
+        try:
+
```
Comment (jaingaurav) [inline src/collectors/systemd/systemd-collector.py]: More of a style thing, please remove this empty line

File: src/collectors/systemd/systemd-collector.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            manager = Manager()
+
+            for service in [s.strip() for s in self.config['services']]:
+
```
Comment (jaingaurav) [inline src/collectors/systemd/systemd-collector.py]: More of a style thing, please remove this empty line

File: src/collectors/systemd/systemd-collector.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                try:
+                    unit = manager.get_unit('%s.service' % service)
+                    self.publish("%s.running" % service, 1
+                                 if str(unit.properties.SubState) == "running"
+                                 else 0)
+                except SystemdError, e:
+                    self.log.exception(e)
+
+        finally:
+
```
Comment (jaingaurav) [inline src/collectors/systemd/systemd-collector.py]: More of a style thing, please remove this empty line

File: src/collectors/systemd/systemd-collector.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
+# coding=utf-8
+
+"""
+Custom collector for systemd process control system
+(http://www.freedesktop.org/wiki/Software/systemd/)
+#### Dependencies
+ * python-systemd
+ * diamond
```
Comment (jaingaurav) [inline src/collectors/systemd/systemd-collector.py]: Diamond doesn't need to spelled out as a dependency here

**Comment by jaingaurav** [comment]
@gmlexx So this looks reasonable however a few issues:

- No test cases
- I am confused about https://github.com/systemd/python-systemd vs https://github.com/wiliamsouza/python-systemd it seems the latter is no longer maintained. In fact the second last commit suggests using the official library. Is it possible to use the official library instead? Unfortunately when going through the docs, I could not see an obvious way to get the service stat information.

**Comment by gmlexx** [comment]
Thank for review. The latter systemd module seems has some leak also.
I will look at the official module.

**Pull request closed without merging**